src/lib/components/ratio/index.svelte

```diff
@@ -1,15 +1,30 @@
 <script lang="ts">
-  import { contrast } from 'colorjs.io/fn';
+  import { contrast, mix } from 'colorjs.io/fn';
 
   import ColorIssues from '$lib/components/ratio/ColorIssues.svelte';
   import Result from '$lib/components/ratio/Result.svelte';
   import ExternalLink from '$lib/components/util/ExternalLink.svelte';
+  import Icon from '$lib/components/util/Icon.svelte';
   import { RATIOS } from '$lib/constants';
   import { bg, fg } from '$lib/stores';
 
-  let ratio = $derived(contrast($bg, $fg, 'WCAG21'));
+  let fgPremultiplied = $derived.by(() => {
+    if ($fg.alpha === 1 || $bg.alpha !== 1) return $fg;
+    return mix($bg, $fg, $fg.alpha, {
+      space: 'srgb',
+      premultiplied: false,
+    });
+  });
+  let ratio = $derived(contrast($bg, fgPremultiplied, 'WCAG21'));
   let displayRatio = $derived(Math.round((ratio + Number.EPSILON) * 100) / 100);
   let pass = $derived(ratio >= RATIOS.AA.Normal);
+  let alphaWarning = $derived.by(() => {
+    if ($bg.alpha !== 1)
+      return 'Alpha is not considered when the background is not opaque.';
```

Comment:
Could we link these two notes directly to the fuller description (expanding if necessary)?

If not that, we could consider adding the fuller explanation in a tooltip/dialog here instead of in the expandable "issues" section.

Reply:
I think linking will be much easier once we're done with https://github.com/oddbird/oddcontrast/pull/223. I'm fine with moving it to a tooltip/dialog, but note we'd need to develop that pattern. I'll defer to @SondraE on what we should do.

Reply:
Correct- that would be great!